azimuth_identity/operator.py

```diff
@@ -153,24 +153,32 @@ async def reconcile_realm(instance: api.Realm, **kwargs):
         LOGGER.info("Updating realm status to PENDING - %s", format_instance(instance))
         await save_instance_status(instance)
     # Derive the realm name from the realm instance
-    realm_name = keycloak.realm_name(instance)
-    # Ensure that the Dex instance for the realm exists
-    dex_client = await dex.ensure_realm_instance(ekclient, instance, realm_name)
+    realm_name = keycloak.realm.realm_name(instance)
     # Create the realm
-    await keycloak.ensure_realm(realm_name)
-    # Make sure that the profile requirements for the realm are set up correctly
-    await keycloak.ensure_user_profile_config(realm_name)
-    # Create and wire up the admins and platform users group
-    await keycloak.ensure_admins_group(realm_name)
-    await keycloak.ensure_platform_users_group(realm_name)
-    # Create and wire up the identity provider for Dex
-    await keycloak.ensure_identity_provider(instance, realm_name, dex_client)
+    await keycloak.realm.ensure_realm(realm_name)
     # Create the groups scope
-    await keycloak.ensure_groups_scope(instance, realm_name)
+    await keycloak.realm.ensure_groups_scope(instance, realm_name)
+    # If required, configure the realm as a tenancy realm
+    # If not, just leave it as a vanilla realm
```

Comment:
I think we should expand this comment with a bit more around the reason why, I am not getting the intent for the difference when the tenancy_id is missing... ah I guess its for the global Azimuth real case? So what is different there, I guess we don't need the DEX bit to forward auth from Azimuth?

Maybe this should say, when its a tenancy real, we add in the DEX system to forward the azimuth sessions, otherwise its an independent realm, such as the one used to authenticate the azimuth session.